src/features/trip_distance.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import os
+
+import joblib
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
+
+from utils import get_model_path
+
+logger = logging.getLogger(__name__)
+
+
+class ModelTripDistanceTransformer(BaseEstimator, TransformerMixin):
+    """
+    Author: Dawid Kowalczyk
+    Date: 7 December 2023
+    Description: Predicting trip_distance based on other features, to have additional input for final model
+    """
+
+    def __init__(self, force_retrain: bool, model_dir: str, model_max_depth: int, model_columns: list):
+        self.force_retrain = force_retrain
+        self.model_dir = model_dir
+        self.model_path = get_model_path() / self.model_dir
+        self.model_name = None
+        self.model = None
+
+        self.model_name = f"trip_distance_model.pkl"
+        self.model_columns = model_columns
+        self.max_depth = model_max_depth
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
```

Comment:
that part I don't like, do you have idea how to better pass X and y? Now I take y from X and then remove redundant columns in 37/38

Reply:
i think that since we are using this model as a feature transformer, this is the proper way